### PR TITLE
[add] website_sale_product_category_seo module

### PR DESCRIPTION
**Pull request by OSguard**
Module to add a individual cms block on top and bottom of category page

### CODE REVIEW

**Comment by gurneyalex** [comment]
Hello @OSguard,

I think there was a typo in your github login in the OCA database. I fixed it, and hopefully the CLAbot will stop flagging your PRs. To be sure, you can commit --amend and push -f this branch. 

Thanks for your contributions to OCA.

File: website_sale_catseo/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Public Product Category CMS Block
+=================================
+
+Make it possible to add a cms block on top and buttom to public product categories in product block and list views.
+
+Contributers
+Michael Karrer 
+Markus Schneider <markus.schneider at initos.com>
```
Comment (gurneyalex) [inline website_sale_catseo/README.md]: Please rename to README.rst : odoo does not understand .md files.

File: website_sale_catseo/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    'summary': """add Public Product Categories CMS Blocks""",
+    'author': "initOS GmbH & Co. KG,OpenAT,Odoo Community Association (OCA)",
+    'website': "http://www.initos.com/",
+    'category': 'website',
+    'version': '0.1',
+    'depends': [
+        'website_sale',
+    ],
+    'installable': True,
+    'css': [
```
Comment (gurneyalex) [inline website_sale_catseo/__openerp__.py]: this key is not used in 8.0

File: website_sale_catseo/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        ],
+    'data': [
+        'views/templates.xml',
+    ],
+    'js': [
```
Comment (gurneyalex) [inline website_sale_catseo/__openerp__.py]: this key is not used in 8.0

File: website_sale_catseo/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Public Product Category CMS Block
+=================================
+
+Make it possible to add a cms block on top and buttom to public product categories in product block and list views.
```
Comment (gurneyalex) [inline website_sale_catseo/README.md]: maybe you could explain what a cms block is? I'm not a specialist, but for me cms == content management system, and it sounds strange here.

**Comment by gurneyalex** [comment]
I don't understand the name of the addon: website_sale_catseo. Does catseo refer to a well known product? or is it category Search Engine Optimisation? the summary and description mention "product category"... Maybe website_sale_product_category_seo would be a better name?

File: website_sale_catseo/README.md (hunk 1 above)
Comment (OSguard) [inline website_sale_catseo/README.md]: on my installation it is working

File: website_sale_product_category_seo/README.rst [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Public Product Category CMS Block
+=================================
+
+Make it possible to add snippets on top and buttom to public product categories in product block and list views. This is simular to magento cms_block feature and you can provide content to optimice your google rank.
```
Comment (hbrunn) [inline website_sale_product_category_seo/README.rst]: simular -> similar
optimice -> optimize

File: website_sale_product_category_seo/views/templates.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+<data>
+
+    <template id="products_categories_cms" inherit_id="website_sale.products" name="Public Category CMS Block">
+        <xpath expr="//div[@id='products_grid']//*[1]" position="before">
+            <div t-if="category">
+                <div t-field="category.cms_top" class=""/>
```
Comment (hbrunn) [inline website_sale_product_category_seo/views/templates.xml]: does the empty class have any special meaning? if not, better remove it

File: website_sale_product_category_seo/views/templates.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+<data>
+
+    <template id="products_categories_cms" inherit_id="website_sale.products" name="Public Category CMS Block">
+        <xpath expr="//div[@id='products_grid']//*[1]" position="before">
+            <div t-if="category">
```
Comment (hbrunn) [inline website_sale_product_category_seo/views/templates.xml:7]: maybe `<div t-if="category and category.cms_top">`?

**Comment by hbrunn** [comment]
Now those fields are only editable via the website view. Would if make sense to also add them to the backend form?

File: website_sale_product_category_seo/views/templates.xml (hunk 7 above)
Comment (OSguard) [inline website_sale_product_category_seo/views/templates.xml:7]: i need the field to edit it in the frontend, if it is empty then i can not fill it

**Comment by OSguard** [comment]
i see no benefit to have edit in the backend. Inline-editing is way-way better then wysiwyg-editor. It is a other philosophy but we should follow this.

**Comment by hbrunn** [comment]
@OSguard thanks for your replies, I see your point.

Please correct the two typos, then I can lift the 'needs fixing' label

**Comment by OSguard** [comment]
just updated

**Comment by hbrunn** [comment]
:+1:

File: website_sale_product_category_seo/README.rst [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Public Product Category CMS Block
+=================================
+
+Make it possible to add snippets on top and buttom to public product categories in product block and list views. This is similar to magento cms_block feature and you can provide content to optimize your google rank.
```
Comment (pedrobaeza) [inline website_sale_product_category_seo/README.rst]: s/buttom/bottom

Comment (pedrobaeza) [inline website_sale_product_category_seo/README.rst]: Don't put Google, but SEO rank

File: website_sale_product_category_seo/README.rst [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+
+Contributers
```
Comment (pedrobaeza) [inline website_sale_product_category_seo/README.rst]: Use OCA's README template: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: website_sale_product_category_seo/__openerp__.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+#
+##############################################################################
+
+{
+    'name': "SEO for Product Categorie",
+    'summary': """add Public Product Categories CMS Blocks""",
+    'author': "initOS GmbH & Co. KG,OpenAT,Odoo Community Association (OCA)",
+    'website': "http://www.initos.com/",
+    'category': 'website',
+    'version': '0.1',
```
Comment (pedrobaeza) [inline website_sale_product_category_seo/__openerp__.py]: You can bump to version 1.0

File: website_sale_product_category_seo/__openerp__.py [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    'depends': [
+        'website_sale',
+    ],
+    'installable': True,
+    'data': [
+        'views/templates.xml',
+    ],
```
Comment (pedrobaeza) [inline website_sale_product_category_seo/__openerp__.py]: Remove this empty section

File: website_sale_product_category_seo/models/product_public_category.py [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+#
+##############################################################################
+
+from openerp import models, fields
+
+
+class product_public_category(models.Model):
+    _inherit = 'product.public.category'
+
+    cms_top = fields.Html()
```
Comment (pedrobaeza) [inline website_sale_product_category_seo/models/product_public_category.py]: Add help attribute indicating the use of these fields

File: website_sale_product_category_seo/views/templates.xml [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+                <div t-field="category.cms_top" id="category_cms_top"/>
```
Comment (pedrobaeza) [inline website_sale_product_category_seo/views/templates.xml:8]: Can you put a placeholder to help to identify the field?

Comment (OSguard) [inline website_sale_product_category_seo/views/templates.xml:8]: odoo core itself adds a placeholder already here. That this placeholder has not information which field or data object it is, is bad but should fix IMO in core.
Adding additional placeholder means that the page looks different in edit mode then in view mode and this is against the concept.
Odoo should implement a hover with information in the core, our placeholder is a bad workaround so this is the reason why i don't add it.

**Comment by OSguard** [comment]
other thinks are changed

**Comment by vilder** [comment]
@OSguard 

Have tested it , works great :+1:

**Comment by vilder** [comment]
Could you add the seo feature also for html title tags in shop categories, with the help of promote button.
Now all the categories have the same title

**Comment by rafaelbn** [comment]
This PR has two validations for a long time...

**Pull request merged**